subsys/net/l2/openthread/Kconfig

```diff
@@ -201,6 +201,67 @@ config OPENTHREAD_DHCP6_SERVER
 	help
 	  Enable DHCPv6 server capability in OpenThread stack
 
+config OPENTHREAD_SLAAC
+	bool "SLAAC support"
+	help
+	  Enable SLAAC capability in OpenThread stack
+
+config OPENTHREAD_ENABLE_SERVICE
+	bool "Service support"
+	help
+	  Enable service capability in OpenThread stack
+
+config OPENTHREAD_RAW
+	bool "Raw Link support"
+	help
+	  Enable raw link support in OpenThread stack
+
+config OPENTHREAD_BORDER_AGENT
+	bool "Border Agent support"
+	help
+	  Enable border agent support in OpenThread stack
+
+config OPENTHREAD_BORDER_ROUTER
+	bool "Border Router support"
+	help
+	  Enable border router support in OpenThread stack
+
+config OPENTHREAD_NCP
+	bool "Network Co-Processor"
```

Comment:
`uart.c` unconditionally depends on ring buffer and interrupt driver UART api, I suggest adding the following here:
```
select RING_BUFFER
select UART_INTERRUPT_DRIVEN
```
Otherwise, the NCP sample does not build in basic configuration (w/o overlay).

Reply:
Added.